### DISCUSSION TITLE
Function names do not support UTF-8/Unicode

### DISCUSSION THREAD

**Discussion by ulughann** [yuin/gopher-lua#442] [General]
Functions or variable names do not support utf8 encoding instead relying on ascii via bytes in go.

eg.
`local ğ = "hi"`
is unusable as well as
`
funşi()
`

**Comment by cloudwindy** [comment]
I don't think the original Lua supports this. And using UTF-8 for function names is a very bad idea.

**Comment by baidwwy** [comment]
Like Lua official, add option?
https://github.com/lua/lua/blob/6baee9ef9d5657ab582c8a4b9f885ec58ed502d0/lctype.c#L20C14-L20C22
